docs/topics/yaml-rule-structure-syntax.adoc

```diff
@@ -0,0 +1,520 @@
+    title: "Short title for the link"
+----
+
+Message can also be templated to include information about the match interpolated through custom variables on the rule.
+
+.Tag actions
+
+The tag action instructs the analyzer to generate one or more tags for the application when a match is found. Each string in the `tag` field can be a comma-separated list of tags. Optionally, you can assign categories to tags.
+
+[source,terminal]
+----
+tag:
+  - "tag1,tag2,tag3"
+  - "Category=tag4,tag5"
+----
+
+_Example_
+
+[source,terminal]
+----
+- ruleID: test-rule
+  when:
+    <CONDITION>
+  tag:
+  - Language=Golang
+  - Env=production
+  - Source Code
+----
+
+A tag can be a string or a `key=val` pair, where the key is treated as a tag category in {ProductShortName}. Any rule that has a tag action is referred to as a “tagging rule” in this document.
+
+Note that issues are not created for rules that contain only tag actions.
+
+
+[id="yaml-rule-conditions_{context}"]
+=== Rule conditions
+
+Each rule has a `when` block, which specifies a condition that needs to be met for {ProductShortName} to perform a certain action.
+
+The `when` block contains one condition, but that condition can have multiple conditions nested under it.
+
+[source,terminal]
+----
+when:
+  <condition>
+    <nested-condition>
+----
+
+{ProductShortName} supports three types of conditions: `provider`, `and` and `or`.
+
+==== Provider conditions
+
+{ProductShortName} supports multi-language source code analysis. Source code search for a specific language is enabled using the `provider` condition. This condition defines a search query for a specific language provider.  The `provider` condition also specifies which of the provider's "capabilities" to use for analyzing the code.
+
+The `provider` condition has the form `<provider_name>.<capability>`:
+
+[source,terminal]
+----
+when:
+  <provider_name>.<capability>
+    <input_fields>
+----
+
+The analyzer currently supports the following `provider` conditions:
+
+* `builtin`
+* `java`
+* `go`
+
+===== `builtin` provider
+
+`builtin` is an internal provider that can analyze various files and internal metadata generated by the engine.
+
+This provider has the following capabilities:
+
+* `file`
+* `filecontent`
+* `xml`
+* `json`
+* `hasTags`
+
+.`file`
+
+The `file` capability enables the provider to search for files in the source code that match a given pattern.
+
+[source,terminal]
+----
+when:
+  builtin.file:
+    pattern: "<regex_to_match_filenames>"
+----
+
+.`filecontent`
+
+The `filecontent` capability enables the provider to search for content that matches a given pattern.
+
+[source,terminal]
+----
+when:
+  builtin.filecontent:
+    filePattern: "<regex_to_match_filenames_to_scope_search>"
+    pattern: "<regex_to_match_content_in_the_matching_files>"
+----
+
+.`xml`
+
+The `xml` capability enables the provider to query XPath expressions on a list of provided XML files. This capability takes 2 input parameters.
+
```

Comment:
Suggestion: "2 input parameters, `xpath` and `filepaths`. Otherwise, the reader is left hanging...